### PR TITLE
Add Self Hosted Charts

### PR DESCRIPTION
**Pull request by ttschampel**
Add self hosted charts =  combination of OpenTDF + Charts for non-public platform services (access pep, configuration service, tagging pdp, sharepoint webhook).

Non-public services are still protected via ghcr authentication, but the charts here are public

### CODE REVIEW

**Review by c3foto — APPROVED** [review]
@ttschampel In general I approve of adding this to the repo but this is a big update and if you need someone to give a closer look at the charts @jfuen10 will be more qualified than me for a second look.

**Comment by ttschampel** [comment]
> @ttschampel In general I approve of adding this to the repo but this is a big update and if you need someone to give a closer look at the charts @jfuen10 will be more qualified than me for a second look.

Sounds good - mostly want to make sure this merge is not bungling anything like the index.yaml for the current charts in main line.

File: sharepoint/templates/istio/gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{{- if and ( .Values.ingress.istio.enabled ) ( not .Values.ingress.istio.existingGateway ) }}
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
```
Comment (strantalis) [inline sharepoint/templates/istio/gateway.yaml:3]: I am not sure this one service requires its own Gateway. Couldn't you have one gateway for all of scp?

Comment (ttschampel) [reply sharepoint/templates/istio/gateway.yaml:3]: Yes, trying to support two modes:  1) As part of the scp cluster 2) external cluster to scp.

You can share the same gateway as scp by setting the .Values.ingress.istio.existingGateway value

File: sharepoint/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          image: {{ .Values.image.repo }}:{{ .Values.image.tag }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.service.port }}
+              protocol: TCP
+          envFrom:
+            {{- if .Values.secretRef }}
```
[the remark this answers is not in the document]
Comment (ttschampel) [reply sharepoint/templates/deployment.yaml:42]: oh, yes. this should be .Values.existingSecret .  change coming

File: index.yaml
[the remark this answers is not in the document]
Comment (ttschampel) [reply index.yaml:1]: I hope so.  We could use some actions in this repo or another repo (private) to do testing before publishing the chart to the repo.

File: scp/charts/configuration/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# OpenTDF Configuration Chart
+
+A Helm chart fo OpenTDF Configuration, a service for providing configuration to applications and other services.
+
+This chart requires [Istio](https://istio.io) Service Mesh to be installed.
+
+This chart has a Redis HA dependency from here: https://github.com/DandyDeveloper/charts/tree/master/charts/redis-ha
```
[the remark this answers is not in the document]
Comment (ttschampel) [reply scp/charts/configuration/README.md]: Removed comment

File: scp/charts/configuration/templates/server/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}-{{ .Values.server.name }}
```
[the remark this answers is not in the document]
Comment (ttschampel) [reply scp/charts/configuration/templates/server/deployment.yaml]: updated

File: scp/charts/configuration/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (strantalis) [reply scp/charts/configuration/README.md]: I think all references to redis for this service need to be removed for this service. Right @pflynn-virtru.

[the remark this answers is not in the document]
Comment (ttschampel) [reply scp/charts/configuration/README.md]: All redis references are now removed

**Pull request merged**